### ISSUE TITLE
DP simulated data has inverted logFC

### ISSUE THREAD

**Issue by ChristophH** [HelenaLC/muscat#125]
Hi,

Thanks for creating and maintaining muscat, and all your hard work that goes into method development and benchmarking!

I think there is a bug/inconsistency in the sign of the logFC of simulated data. When simulated with the 'dp' category, the logFC is applied in the opposite direction as, for example, with the 'de' or 'dm' category.

In .sim() you can see how either '-lfc', or 'lfc' is used with group 1

https://github.com/HelenaLC/muscat/blob/10cd3262bd5bc45930d62dc8f7d8b97629c3bbe4/R/utils-simData.R#L329-L352

But maybe I'm missing something.

Cheers,
Christoph

**Comment by HelenaLC** [comment]
Yeah, so it’s admittedly been a while since writing this, but looking at the code, I think the reasoning is at follows…
- we generally assume the prop. of DP genes (dp) to be low, say 10%.
- if all logFCs were positive, 10% in group 1 would be “up” and 90% in group 2 would be down, so group 1 would be relatively higher. 
- since logFCs are drawn symmetrically, the same holds vice versa, and that the sign doesn’t really matter, which distinguishes this case from others
- so, the code is just how this has been implemented, but I understand the confusion
- note that if dp=0.5, both groups get the exact same effect
…hope that makes sense?

**Comment by ChristophH** [comment]
Thanks for the explanation. However, I was just trying to validate that behavior using the .sim function with some toy parameters, but could not reproduce the DP density curves. 
<img width="333" alt="image" src="https://github.com/HelenaLC/muscat/assets/3731029/3bc05b7a-7880-476b-8dc8-3abc523929ab"> [Crowell et al., Nat Com, 2020]

For DE and DM the curves look as expected. But for DP, only group A shows the bimodal distribution. As a result, the observed logFC is always negative, even
[test_muscat.R.pdf](https://github.com/HelenaLC/muscat/files/12339977/test_muscat.R.pdf)
 when the input logFC is positive. 
<img width="333" alt="image" src="https://github.com/HelenaLC/muscat/assets/3731029/83c8dfca-c972-4924-b465-07b40ecc9503">

(For details see attached R notebook).

I understand that .sim is an internal function, but I'm going down this rabbit hole as I'm trying to understand why my muscat generated data for DP genes does not look as expected.

**Comment by ChristophH** [comment]
Short recap:

1. Genes generated under the DP model show unexpected logFC (observed logFC is almost always opposite direction of the logFC of the gene_info table in the simulation metadata)
2. Genes generated under the DP model do not show bimodal distribution in both groups (just in one group, this explains the issue above)

To me, this seems like unexpected behavior (a bug?), or am I doing it wrong? Please advice.
